src/cortxfs/cortxfs_fops.c

```diff
@@ -159,17 +159,25 @@ static inline ssize_t __cfs_write(struct cfs_fs *cfs_fs, cfs_cred_t *cred,
 				  size_t count, off_t offset)
 {
 	int rc;
-	struct stat stat;
 	dstore_oid_t oid;
+	struct stat *stat = NULL;
+	struct cfs_fh *fh = NULL;
 	struct dstore *dstore = dstore_get();
-	struct kvnode node = KVNODE_INIT_EMTPY;
 	struct dstore_obj *obj = NULL;
 
+	dassert(cfs_fs && cred && fd && buf);
 	dassert(dstore);
 
+	/* TODO:Temp_FH_op - to be removed
+	 * Should get rid of creating and destroying FH operation in this
+	 * API when caller pass the valid FH instead of inode number
+	 */
+	RC_WRAP_LABEL(rc, out, cfs_fh_from_ino, cfs_fs, &fd->ino, &fh);
+	stat = cfs_fh_stat(fh);
+
 	RC_WRAP_LABEL(rc, out, cfs_ino_to_oid, cfs_fs, &fd->ino, &oid);
```

Comment:
Once file handle support is in, the kv store operations in cfs_ino_to_oid will continue to be a bottleneck till we cache the oid as well..Can we put a comment here mentioning the same i.e PERF_IMPROVEMENT
And the new function i guess will be cfs_fh_to_oid?

Reply:
cfs_fh structure has the TODO for it to be done as a next improvement and the JIRA ticket EOS-8934 where FH design details are mentioned.